crates/uv-workspace/src/workspace.rs

```diff
@@ -458,8 +458,65 @@ impl Workspace {
             Some(workspace.install_path.join(path))
         }
 
-        // TODO(zanieb): Warn if `VIRTUAL_ENV` is set and does not match
-        from_environment_variable(self).unwrap_or_else(|| self.install_path.join(".venv"))
+        // Resolve the `VIRTUAL_ENV` variable, if any.
+        fn from_virtual_env_variable() -> Option<PathBuf> {
+            let value = std::env::var_os("VIRTUAL_ENV")?;
+
+            if value.is_empty() {
+                return None;
+            };
+
+            let path = PathBuf::from(value);
+            if path.is_absolute() {
+                return Some(path);
+            };
+
+            // Resolve the path relative to current directory.
+            // Note this differs from `UV_PROJECT_ENVIRONMENT`
+            Some(CWD.join(path))
+        }
+
+        // Attempt to check if the two paths refer to the same directory.
+        fn is_same_dir(left: &Path, right: &Path) -> Option<bool> {
+            // First, attempt to check directly
+            if let Ok(value) = same_file::is_same_file(left, right) {
+                return Some(value);
+            };
+
+            // Often, one of the directories won't exist yet so perform the comparison up a level
+            if let (Some(left_parent), Some(right_parent), Some(left_name), Some(right_name)) = (
```

Comment:
A little awkward, but important for cases where `.venv` is being created — it'd be weird not to warn on the first invocation but warn on later ones.

Reply:
Is there any value to the previous check when this one's already here? This just feels like a more general version of the previous one. The only thing I can think of is that this one will trigger on case-insensitive filesystems if the casing differs, while I don't think that `is_same_file` will.

Reply:
I'm not sure honestly. Seems safe to keep it.

cc @BurntSushi for a post-merge review if you have time.